### DISCUSSION TITLE
Baseline for future un-joined saving session events.

### DISCUSSION THREAD

**Discussion by deltamelter** [BottlecapDave/HomeAssistant-OctopusEnergy#1131] [Ideas]
I know the whole OE Saving Sessions and their API use is only half baked this season but I was trying to add saving session visualization to some of my plotly graphs.  I've added a plot trace to display a rectangle (bar) shape covering the start and duration and basic details like reward/kWh saved (currently from joined_events).  I was hoping to also indicate the current baseline for the event to indicate whether it is worth opting in at all and how hard we would have to try to make a saving.
Currently the integration only shows available and joined events and the baseline entity only seems to relate to joined events.  (I know they haven't used this properly yet so have yet to see this in practice).  It would be great if the future events could have the baseline calculated so that for dayahead (if they ever happen again) the baseline could be improved and for same day events the baseline would let us know if it was worth bothering. 
completed joined events in the API now indicate the OE calculated baseline, consumption and reward which would be good to make available to plotly too 

```
      joinedEvents {
        status
        startAt
        rewardGivenInOctoPoints
        resultsStatus
        resultsSetAt
        percentageSaved
        netReductionPctRank
        eventStatus
        eventId
        energySavedInKwh
        endAt
        consumptionDeltaKwh
        co2SavedInGrams
        baselineConsumptionDeltaKwh
        avgCustomerRemunerationPounds
      }
      
```

**Comment by BottlecapDave** [comment]
Hello. I'm not looking at doing anything around saving sessions until they've finalised what they're doing with the API (either adapting the current API or creating a new one).

> Currently the integration only shows available and joined events and the baseline entity only seems to relate to joined events

This is correct and was a design decision. 

The OE API doesn't currently have any endpoint for requesting this calculation bespoke, and originally didn't have anything on the joined events endpoint either. It looks like the joined events endpoint your overall baseline, whereas it is actually calculated on thirty minute segments over the session. This means that if you don't save during one thirty minute session you still have a chance in the next thirty minute segment. This looks to still be the case under "How do you work out my typical energy use?" at https://octopus.energy/blog/saving-sessions-faqs.

Currently, all of the required data is built up over time and cached for baseline calculations locally. It was only done on joined events to avoid unneccesary processing on this data, as it can be expensive to calculate, more so if there are gaps in the cached data as it has to request that data from the OE APIs.

Changing this behaviour would result in a possible breaking change with the baseline blueprint as it would then require the blueprint to know if the user was in an active session. I'll revist this behaviour when I know more about the future of this API.

> completed joined events in the API now indicate the OE calculated baseline, consumption and reward which would be good to make available to plotly too

If this API continues to be used or is available in the new API, I'll have a look at exposing some more of the values.

In the future, this sort of discussion should be raised as a feature request as it allows me to track it better and also allows other people to vote on the change.
